index.ts

```diff
@@ -80,21 +75,45 @@ export const typescriptPaths = ({
 };
 
 const getTsConfig = (configPath?: string): TsConfig => {
-	const defaults: TsConfig = { compilerOptions: {}, outDir: '.' };
-
-	if (!configPath) {
+	const defaults: TsConfig = { outDir: '.' };
+	if (typeof configPath !== 'string') {
 		return defaults;
 	}
 
-	const configJson = sys.readFile(configPath);
+	// Define a host object that implements ParseConfigFileHost.
+	// The host provides file system operations and error handling for parsing the configuration file.
+	const host: ts.ParseConfigFileHost = {
+		fileExists: ts.sys.fileExists,
+		readFile: ts.sys.readFile,
+		readDirectory: ts.sys.readDirectory,
+		useCaseSensitiveFileNames: ts.sys.useCaseSensitiveFileNames,
+		getCurrentDirectory: ts.sys.getCurrentDirectory,
+		onUnRecoverableConfigFileDiagnostic: (diagnostic) => {
+			console.error(
+				'Unrecoverable error in config file:',
+				diagnostic.messageText,
+			);
+			process.exit(1);
+		},
+	};
 
-	if (!configJson) {
-		return defaults;
+	// Read in tsconfig.json
+	const parsedCommandLine = ts.getParsedCommandLineOfConfigFile(
```

Comment:
This might be a lot slower than what happened before, so probably it makes sense to memoize the `getTsConfig` function for performance. I can add this (hopefully i'll find some time to properly go through this PR in the next few weeks).